_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_vaapi.cpp

```diff
@@ -445,6 +458,41 @@ mfxStatus SetFrameRate(
     return MFX_ERR_NONE;
 }
 
+mfxStatus SetMaxSliceSize(
+    MfxVideoParam const & par,
+    VADisplay    vaDisplay,
+    VAContextID  vaContextEncode,
+    VABufferID & maxSliceSizeBuf_id)
+{
+    mfxStatus sts = CheckAndDestroyVAbuffer(vaDisplay, maxSliceSizeBuf_id);
+    MFX_CHECK_WITH_ASSERT(sts == MFX_ERR_NONE, MFX_ERR_DEVICE_FAILED);
```

Comment:
Direct call to vaDestroyBuffer would be more organic here.

Reply:
Agree with @lakulako . I don't see any reason of using function CheckAndDestroyVAbuffer, especially in this place.

Function CheckAndDestroyVAbuffer only call vaDestroyBuffer and check VA return status - if it not success -> return immediately MFX_ERR_DEVICE_FAILED with assert, that's all.
In current place right after calling CheckAndDestroyVAbuffer you check again returning status and return immediately MFX_ERR_DEVICE_FAILED with assert. Why ? I don't see any benefits of this.

As result, we have a pretty good function vaDestroyBuffer and I don't understand why we need to add new level of calling stack with wrapper function which do nothing and double check of the same status. Also, I don't see any other places where this function is used. Or I missed something ?

Reply:
This function is used and was first introduced in AVC code.

Ok, to get rid of this function and use ```vaDestroy``` we need to make sure that this function is resistant across invalid input, since ```CheckAndDestroyVAbuffer``` check buffer id as well.
Looking here 
https://github.com/intel/libva/blob/e4cd982197a4f0131a6a5a43dd83f69f0a724db7/va/va.c#L1393 
and here
 https://github.com/intel/media-driver/blob/8e41388b5d17c80b126cc7f36505b090f3b1942d/media_driver/linux/common/ddi/media_libva.cpp#L3035
we can say that it should be.
```CheckAndDestroyVAbuffer``` is also set invalid id to buffer id, so we need to pull this assignment out of the wrapper and do on this level (because if subsequent call to ```vaCreateBuffer``` fails, we will have valid ID for deleted buffer - unwelcome situation).
Saying this I see reasons to use ```CheckAndDestroyVAbuffer``` wrapper